src/main.rs

```diff
@@ -97,34 +97,32 @@ fn main() {
     loop {
         db.parse(start.clone(), data.clone());
 
-        let mut types = vec![DiagnosticClass::Parse];
-
         if !matches.is_present("discouraged") {
-            types.push(DiagnosticClass::Scope);
+            db.scope_pass();
         }
```

Comment:
I still have an issue with this line.
If the end user runs the command line tool with the `-verify` option, `scope_pass` will be run anyway.
In other cases, the passes are run lazily, once, whenever needed : I don't see why we need to force the scope pass here specifically.
So I believe we could simply get rid of those 3 lines.

Reply:
This is basically a holdover from the version before `discouraged` when it was always run unconditionally. In particular if you don't select *any* passes with command line options then the default behavior is to get errors from the scope pass.

Reply:
Yes, then I think I would prefer with an option to actively request the scope pass, like for others.

Reply:
I think I'm just going to remove it. It's kind of hard to explain what the scope pass is for users, and there isn't really a good reason to use the option. If you want full verification you will use the verify pass, and if you don't then the minimal parse-only pass is sufficient. Anything else will just do the passes it needs anyway.